tests/end_to_end/tests.py

```diff
@@ -80,9 +88,122 @@ def step_token_init_keys(self):
             sign_key_label = security_server['name'] + '-default-sign-key'
             assert str(response[0].keys[0].label) == auth_key_label
             assert str(response[0].keys[1].label) == sign_key_label
+            self.revoke_api_key(security_server)
+
+    def step_cert_download_csrs(self):
+        with XRDSSTTest() as app:
+            cert_controller = CertController()
+            cert_controller.app = app
+            for security_server in self.config["security_server"]:
+                ss_configuration = cert_controller.initialize_basic_config_values(security_server, self.config)
+                result = cert_controller.remote_download_csrs(ss_configuration, security_server)
+                assert len(result) == 2
+                assert result[0].fs_loc != result[1].fs_loc
+
+                self.revoke_api_key(security_server)
+
+                return [
+                    ('sign', next(csr.fs_loc for csr in result if csr.key_type == 'SIGN')),
+                    ('auth', next(csr.fs_loc for csr in result if csr.key_type == 'AUTH')),
+                ]
+
+    def step_acquire_certs(self, downloaded_csrs):
+        tca_sign_url = find_test_ca_sign_url(self.config['security_server'][0]['configuration_anchor'])
+        cert_files = []
+        for down_csr in downloaded_csrs:
+            cert = perform_test_ca_sign(tca_sign_url, down_csr[1], down_csr[0])
+            cert_file = down_csr[1] + ".signed.pem"
+            cert_files.append(cert_file)
+            with open(cert_file, "w") as out_cert:
+                out_cert.write(cert)
+
+        return cert_files
+
+    def apply_cert_config(self, signed_certs):
+        self.config['security_server'][0]['certificates'] = signed_certs
+
+    def revoke_api_key(self, security_server):
```

Comment:
I really do not like this, the presence of this function and the many places where it is called from seems rather detrimental to readability here. Could it possibly better to just:
  a) register API key for tests and use that
  b) use the ``tearDown()`` method?
  c) as a bonus for more testing, could the application revocation method be used / tested here